tests/client/test_event_hooks.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+import httpx
+
+
+def test_event_hooks(server):
```

Comment:
It seems we're not testing for "request hooks are called on redirect requests", which makes me think the decision to put request hooks at the top of `._send_handling_auth()` was not entirely deliberate? :-)

Any case, whichever we decide I guess we'd need to add explicit tests for what we expect to happen as requests and responses go through the send-redirects-auth-(retries soon) stack?